app/src/main/java/in/ac/bits_pilani/goa/ard/utils/Browser.java

```diff
@@ -0,0 +1,370 @@
+
+    /**
+     * Key for pending intent of menu item.
+     */
+    private static final String KEY_PENDING_INTENT = "browser.menuItems.pendingIntent";
+
+    /**
+     * Key for icon of action button.
+     */
+    private static final String KEY_ACTION_BUTTON_ICON = "browser.actionButton.icon";
+
+    /**
+     * Key for description of action button (for accessibility).
+     */
+    private static final String KET_ACTION_BUTTON_DESC = "browser.actionButton.description";
+
+    /**
+     * Key for tint of action button.
+     */
+    private static final String KEY_ACTION_BUTTON_TINT = "browser.actionButton.tint";
+
+    /**
+     * Key for intent of action button.
+     */
+    private static final String KEY_ACTION_BUTTON_INTENT = "browser.actionButton.intent";
+
+    /**
+     * Host activity.
+     */
+    private final Activity hostActivity;
+
+    /**
+     * Background color for the toolbar. This is an int that specifies a
+     * {@link Color}, not a resource id.
+     */
+    private int toolBarColor;
+
+    /**
+     * Close button icon for the custom tab.
+     */
+    private Bitmap closeButtonIcon;
+
+    /**
+     * Starting animation for browser tab (when url is launched).
+     */
+    private int startAnimEnterResId;
+
+    /**
+     * Exit animation for activity (when url is launched).
+     */
+    private int startAnimExitResId;
+
+    /**
+     * Starting animation for activity (when browser tab is closed).
+     */
+    private int exitAnimEnterResId;
+
+    /**
+     * Exit animation for browser tab (when browser tab is closed).
+     */
+    private int exitAnimExitResId;
+
+    /**
+     * Should url bar hide as the user scrolls down on the page.
+     */
+    private boolean urlBarHiding;
+
+    /**
+     * Should the title of web page be shown.
+     */
+    private boolean titleEnabled;
+
+    /**
+     * Menu items. Each item is a bundle having
+     * a string title {key: KEY_MENU_ITEM_TITLE}
+     * and a pending intent (key: KEY_PENDING_INTENT)
+     */
+    private ArrayList<Bundle> menuItems;
+
+    /**
+     * Whether to include share menu item.
+     */
+    private boolean shareInMenu;
+
+    /**
+     * CustomTabsIntent object.
+     */
+    private CustomTabsIntent customTabsIntent;
+
+    /**
+     * Bundle ActionButton.
+     */
+    private Bundle actionButton;
+
+    /**
+     * Should action button be enabled.
+     */
+    private boolean actionButtonEnabled;
+
+    /**
+     * Constructor initiates parameters to default values.
+     * @param hostActivity activity from which wab tab is launched.
+     */
+    public Browser(final Activity hostActivity) {
+        this.hostActivity = hostActivity;
+        setToolBarColor(ActivityCompat.getColor(hostActivity, R.color.customTabToolbarColor));
+        setCloseButtonIcon(getBitmapFromVectorDrawable(hostActivity, R.drawable.ic_arrow_back_white_24dp));
+        setStartAnimation(R.anim.slide_in_right, R.anim.slide_out_left);
```

Comment:
@Rushi98 My opinion is to have the animation as Google Now has it..it comes from the bottom to up. Left right animations do not seem proper
@yashsharan @shreyasnbhat Any thoughts?